Sources/Tracing/SpanProtocol.swift

```diff
@@ -26,7 +26,7 @@ import struct Dispatch.DispatchWallTime
 /// Creating a `Span` is delegated to a ``Tracer`` and end users should never create them directly.
 ///
 /// - SeeAlso: For more details refer to the [OpenTelemetry Specification: Span](https://github.com/open-telemetry/opentelemetry-specification/blob/v0.7.0/specification/trace/api.md#span) which this type is compatible with.
-public protocol Span: AnyObject, _SwiftTracingSendableSpan {
```

Comment:
Something that has just come to my mind, if we try to make not using a tracer as noop as possible we should lift the `AnyObject` requirement here. This way we can implement Noops with empty spans.

Reply:
Not sure that makes sense... Semantically they really are reference types.

You also wanted to return _concrete types_ making it not possible to return `NoopSpan` "sometimes" and the "real one" usually. since that's different types.

So I guess the only trick this could pull then is to have a struct or enum that is "the noop one or the real one that contains a class `box` with the actual impl. I'm assuming we'd keep the methods not mutating even then, since value semantics are _wrong_ for spans.

Reply:
> Not sure that makes sense... Semantically they really are reference types.

100% agree. In NIO we sometimes use structs that wrap reference types and we document heavily that they have reference semantics. 

> to have a struct or enum that is "the noop one or the real one that contains a class box with the actual impl. I'm assuming we'd keep the methods not mutating even then, since value semantics are wrong for spans.

Yes! 100% this.

Reply:
This "let's pretend `nonmutating`" won't work, I feel like I've been through this multiple times already if I'm honest.

I won't do this dance in this PR.

Things break e.g. here:

```swift
        let tracer: any TracerProtocol = InstrumentationSystem.tracer

        var baggage = Baggage.topLevel
        InstrumentationSystem.instrument.extract(request.headers, into: &baggage, using: HTTPHeadersExtractor())

        let span = tracer.startSpan("GET \(request.path)", baggage: baggage)

        let response = self.catchAllHandler(span.baggage, request, self.client)
        span.attributes["http.status"] = response.status
```

Reply:
solved in https://github.com/apple/swift-distributed-tracing/pull/94